### ISSUE TITLE
Grid+GAM: Reproducible nondeterminism in Grid search

### ISSUE THREAD

**Issue by tomasfryda** [h2oai/h2o-3#16662]
In `rel-3.46.0` GAM fails `pyunit_PUBDEV_7860_cartesian_gridsearch_subspaces_TP_CS_MS_binomial_dual_mode.py` test. 

When investigating I found out that the test fails on the first run after starting h2o. Any subsequent run seems to be ok.

I managed to make the hyperspace smaller while ensuring high probability of reproducing this issue:
```python
def test_gridsearch():
    h2o_data = h2o.import_file(path = pyunit_utils.locate("smalldata/gam_test/synthetic_20Cols_binomial_20KRows.csv"))
    h2o_data['response'] = h2o_data['response'].asfactor()
    h2o_data['C3'] = h2o_data['C3'].asfactor()
    h2o_data['C7'] = h2o_data['C7'].asfactor()
    h2o_data['C8'] = h2o_data['C8'].asfactor()
    h2o_data['C10'] = h2o_data['C10'].asfactor()
    names = h2o_data.names
    myY = "response"
    myX = names.remove(myY)
    search_criteria = {'strategy': 'Cartesian'}
    hyper_parameters = {'lambda': [1, 2],
                            'subspaces': [{'scale': [[0.001], [0.0002]],
                                            'bs':[[1], [0]],
                                           'gam_columns': [[["c_0"]], [["c_1"]]]},
                                          {'bs':[[3, 1,], [0, 1,]],
                                           'gam_columns': [["c_1", ["c_2", "c_3"]]]}]}                                                   
    
    h2o_model = H2OGridSearch(H2OGeneralizedAdditiveEstimator(family="binomial", keep_gam_cols=True, seed=1),
                              hyper_params=hyper_parameters, search_criteria=search_criteria)
    h2o_model.train(x = myX, y = myY, training_frame = h2o_data)
    h2o_model2 = H2OGridSearch(H2OGeneralizedAdditiveEstimator(family="binomial", keep_gam_cols=True, seed=1),
                              hyper_params=hyper_parameters, search_criteria=search_criteria)
    h2o_model2.train(x = myX, y = myY, training_frame = h2o_data)
    # compare two models by checking their coefficients.  They should be the same
    for index in range(0, len(h2o_model)):
        model1 = h2o_model[index]
        model2 = h2o_model2[index]
        if len(model1.coef()) != len(model2.coef()):
            print(f"{index=}")
            print(model1.coef())
            print(model2.coef())
        pyunit_utils.assertEqualCoeffDicts(model1.coef(), model2.coef(), tol=1e-6)

```

Difference between the first and second run is:
```python
# a is the first run
# b is the second run
In [2]: for k in a.keys():
   ...:     if k not in b:
   ...:         print(k)
c_1_0_center_cs__0
c_1_0_center_cs__1
c_1_0_center_cs__2
c_1_0_center_cs__3
c_1_0_center_cs__4
c_1_0_center_cs__5
c_1_0_center_cs__6
c_1_0_center_cs__7
c_1_0_center_cs__8

In [3]: for k in b.keys():
   ...:     if k not in a:
   ...:         print(k)
c_1_3_center_0
c_1_3_center_1
``` 
---
After some time I found out, that the error above is due to different ordering of the trained models. The models have the same metrics (at least the visible decimals of the metric) and for some reason are ordered differently. I suspect it could be caused by subspaces which also cause wrong rendering of "Hyper-Parameter Search Summary" (see the screenshot for the two grids (`h2o_model` and `h2o_model2`).

<img width="1432" height="834" alt="Image" src="https://github.com/user-attachments/assets/820dcf98-3b1a-42f5-b390-51fd2ac95e91" />

---

In Java the two models have different logloss in Java (diff=1.3877787807814457E-17) (but not in python likely due to rounding error).

So it looks like the subspaces are an unrelated error.

---

There might be a bug in thin plane since I'm getting different results of thin plane for that one particular GAM:

<img width="2612" height="563" alt="Image" src="https://github.com/user-attachments/assets/76a5886a-fe91-4d2a-adcd-c14e398bf30c" />

**Comment by hutch3232** [comment]
Two related issues I reported awhile back:
https://github.com/h2oai/h2o-3/issues/6563#issuecomment-1657175318
https://github.com/h2oai/h2o-3/issues/6562#issuecomment-1657188568
